lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+            returned: always
+            type: string
+        stored_bytes:
+            description: The number of bytes stored.
+            returned: always
+            type: string
+        kms_key_id:
+            description: The Amazon Resource Name (ARN) of the CMK to use when encrypting log data.
+            returned: always
+            type: string
+'''
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import _camel_to_snake, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_log_group(client, log_group_name, kms_key_id, tags, retention, module):
+    try:
+        request = {
+            'logGroupName': log_group_name
+        }
+        if kms_key_id:
+            request['kmsKeyId'] = kms_key_id
+        if tags:
+            request['tags'] = tags
+
+        client.create_log_group(**request)
+
+        if retention:
+            input_retention_policy(client=client,
+                                   log_group_name=log_group_name,
+                                   retention=retention, module=module)
+
+        desc_log_group = describe_log_group(client=client,
+                                            log_group_name=log_group_name,
+                                            module=module)
+
+        if 'logGroups' in desc_log_group:
+            for i in desc_log_group['logGroups']:
+                if log_group_name == i['logGroupName']:
+                    return i
+                else:
+                    module.fail_json(msg="The aws CloudWatchLogs log group was not created. \n please try again!")
+
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+
+
+def input_retention_policy(client, log_group_name, retention, module):
+    try:
+        permited_values = [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]
+
+        if retention in permited_values:
+            response = client.put_retention_policy(logGroupName=log_group_name,
+                                                   retentionInDays=retention)
+        else:
+            delete_log_group(client=client, log_group_name=log_group_name, module=module)
+            module.fail_json(msg="Invalid retention value. Valid values are: [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]")
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.response, exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+
+
+def delete_log_group(client, log_group_name, module):
+    try:
```

Comment:
You can just have the try/except around the delete_log_group() section since describe_log_group() has its own handling.